File: app/src/main/java/vn/loitp/app/activity/animation/MenuAnimationActivity.java
package vn.loitp.app.activity.animation;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import loitp.basemaster.R;
import vn.loitp.app.activity.animation.activitytransition.Animation1Activity;
import vn.loitp.app.activity.animation.animationview.AnimationViewActivity;
import vn.loitp.app.activity.animation.basictransition.BasicTransition0Activity;
import vn.loitp.app.activity.animation.basictransitionactivity.SceneTransitionBasicActivity;
import vn.loitp.app.activity.animation.basictransitionfrm.BasicTransitionActivity;
import vn.loitp.app.activity.animation.confetti.ConfettiMenuActivity;
import vn.loitp.app.activity.animation.expectanim.ExpectAnimActivity;
import vn.loitp.app.activity.animation.flyschool.FlySchoolActivity;
import vn.loitp.app.activity.animation.overscroll.OverScrollActivity;
import vn.loitp.app.activity.animation.shadowviewhelper.ShadowViewHelperActivity;
import vn.loitp.core.base.BaseFontActivity;
import vn.loitp.core.utilities.LActivityUtil;

public class MenuAnimationActivity extends BaseFontActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        findViewById(R.id.bt_animation_view).setOnClickListener(this);
        findViewById(R.id.bt_over_scroll).setOnClickListener(this);
        findViewById(R.id.bt_fly_school).setOnClickListener(this);
        findViewById(R.id.bt_activity_transition).setOnClickListener(this);
        findViewById(R.id.bt_shadowview_helper).setOnClickListener(this);
        findViewById(R.id.bt_expect_anim).setOnClickListener(this);
        findViewById(R.id.bt_animation_confetti).setOnClickListener(this);
        findViewById(R.id.bt_basic_transition_frm).setOnClickListener(this);
        findViewById(R.id.bt_basic_transition_activity).setOnClickListener(this);
        findViewById(R.id.bt_basic_transition).setOnClickListener(this);
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_menu_animation;
    }

    @Override
    public void onClick(View v) {
        Intent intent = null;
        switch (v.getId()) {
            case R.id.bt_animation_view:
                intent = new Intent(activity, AnimationViewActivity.class);
                break;
            case R.id.bt_over_scroll:
                intent = new Intent(activity, OverScrollActivity.class);
                break;
            case R.id.bt_fly_school:
                intent = new Intent(activity, FlySchoolActivity.class);
                break;
            case R.id.bt_activity_transition:
                intent = new Intent(activity, Animation1Activity.class);
                break;
            case R.id.bt_shadowview_helper:
                intent = new Intent(activity, ShadowViewHelperActivity.class);
                break;
            case R.id.bt_expect_anim:
                intent = new Intent(activity, ExpectAnimActivity.class);
                break;
            case R.id.bt_animation_confetti:
                intent = new Intent(activity, ConfettiMenuActivity.class);
                break;
            case R.id.bt_basic_transition_frm:
                intent = new Intent(activity, BasicTransitionActivity.class);
                break;
            case R.id.bt_basic_transition_activity:
                intent = new Intent(activity, SceneTransitionBasicActivity.class);
                break;
            case R.id.bt_basic_transition:
                intent = new Intent(activity, BasicTransition0Activity.class);
                break;
        }
        if (intent != null) {
            startActivity(intent);
            LActivityUtil.tranIn(activity);
        }
    }
}

File: app/src/main/java/vn/loitp/app/activity/animation/MenuAnimationActivity.kt
package vn.loitp.app.activity.animation

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.core.view.isVisible
import com.loitpcore.annotation.IsAutoAnimation
import com.loitpcore.annotation.IsFullScreen
import com.loitpcore.annotation.LogTag
import com.loitpcore.core.base.BaseFontActivity
import com.loitpcore.core.utilities.LActivityUtil
import com.loitpcore.core.utilities.LUIUtil
import kotlinx.android.synthetic.main.activity_menu_animation.*
import vn.loitp.app.R
import vn.loitp.app.activity.animation.activityTransition.Animation1Activity
import vn.loitp.app.activity.animation.activityTransitionReveal.RevealActivity1
import vn.loitp.app.activity.animation.androidParticles.AndroidParticlesActivity
import vn.loitp.app.activity.animation.animatedStarsView.AnimatedStarsViewActivity
import vn.loitp.app.activity.animation.animationView.AnimationViewActivity
import vn.loitp.app.activity.animation.basicTransition.BasicTransition0Activity
import vn.loitp.app.activity.animation.basicTransitionActivity.SceneTransitionBasicActivity
import vn.loitp.app.activity.animation.basicTransitionFrm.BasicTransitionActivity
import vn.loitp.app.activity.animation.elasticViews.ElasticActivity
import vn.loitp.app.activity.animation.fadeOutParticle.FadeOutParticleActivity
import vn.loitp.app.activity.animation.flySchool.FlySchoolActivity
import vn.loitp.app.activity.animation.konfetti.KonfettiActivity
import vn.loitp.app.activity.animation.lottie.MenuLottieActivity
import vn.loitp.app.activity.animation.morphTransitions.MorphTransitionsMainActivity
import vn.loitp.app.activity.animation.overScroll.OverScrollActivity
import vn.loitp.app.activity.animation.pulsingView.PulsingViewActivity
import vn.loitp.app.activity.animation.shadowViewHelper.ShadowViewHelperActivity
import vn.loitp.app.activity.animation.valueAnimator.ValueAnimatorActivity

@LogTag("MenuAnimationActivity")
@IsFullScreen(false)
@IsAutoAnimation(true)
class MenuAnimationActivity : BaseFontActivity(), View.OnClickListener {

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_menu_animation
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setupViews()
    }

    private fun setupViews() {
        lActionBar.apply {
            LUIUtil.setSafeOnClickListenerElastic(
                view = this.ivIconLeft,
                runnable = {
                    onBaseBackPressed()
                }
            )
            this.ivIconRight?.isVisible = false
            this.tvTitle?.text = MenuAnimationActivity::class.java.simpleName
        }
        btAnimationView.setOnClickListener(this)
        btOverScroll.setOnClickListener(this)
        btFlySchool.setOnClickListener(this)
        btActivityTransition.setOnClickListener(this)
        btShadowViewHelper.setOnClickListener(this)
        btBasicTransitionFrm.setOnClickListener(this)
        btBasicTransitionActivity.setOnClickListener(this)
        btBasicTransition.setOnClickListener(this)
        btLottie.setOnClickListener(this)
        btValueAnimator.setOnClickListener(this)
        btElasticView.setOnClickListener(this)
        btMorphTransitions.setOnClickListener(this)
        btActivityTransitionReveal.setOnClickListener(this)
        btPulsingView.setOnClickListener(this)
        btAndroidParticlesActivity.setOnClickListener(this)
        btKonfetti.setOnClickListener(this)
        btFadeOutParticleActivity.setOnClickListener(this)
        btAnimatedStarsView.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        val intent = when (v) {
            btAnimationView -> {
                Intent(this, AnimationViewActivity::class.java)
            }
            btOverScroll -> {
                Intent(this, OverScrollActivity::class.java)
            }
            btFlySchool -> {
                Intent(this, FlySchoolActivity::class.java)
            }
            btActivityTransition -> {
                Intent(this, Animation1Activity::class.java)
            }
            btShadowViewHelper -> {
                Intent(this, ShadowViewHelperActivity::class.java)
            }
            btBasicTransitionFrm -> {
                Intent(this, BasicTransitionActivity::class.java)
            }
            btBasicTransitionActivity -> {
                Intent(this, SceneTransitionBasicActivity::class.java)
            }
            btBasicTransition -> {
                Intent(this, BasicTransition0Activity::class.java)
            }
            btLottie -> {
                Intent(this, MenuLottieActivity::class.java)
            }
            btValueAnimator -> {
                Intent(this, ValueAnimatorActivity::class.java)
            }
            btElasticView -> {
                Intent(this, ElasticActivity::class.java)
            }
            btMorphTransitions -> {
                Intent(this, MorphTransitionsMainActivity::class.java)
            }
            btActivityTransitionReveal -> {
                Intent(this, RevealActivity1::class.java)
            }
            btPulsingView -> {
                Intent(this, PulsingViewActivity::class.java)
            }
            btAndroidParticlesActivity -> {
                Intent(this, AndroidParticlesActivity::class.java)
            }
            btKonfetti -> {
                Intent(this, KonfettiActivity::class.java)
            }
            btFadeOutParticleActivity -> {
                Intent(this, FadeOutParticleActivity::class.java)
            }
            btAnimatedStarsView -> {
                Intent(this, AnimatedStarsViewActivity::class.java)
            }
            else -> null
        }
        intent?.let {
            startActivity(it)
            LActivityUtil.tranIn(this)
        }
    }
}
